nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/log/LogStreamManager.java

```diff
@@ -140,29 +136,39 @@ public RecordFilter getFilter(String stream) {
     protected void initStreams(Topology topology, Settings settings) {
         log.debug("Initializing streams");
         topology.streamsSet().forEach(streamName -> {
-            if (logManager.exists(streamName)) {
+            if (settings.isExternal(streamName)) {
+                return;
+            }
+            if (!logManager.exists(streamName)) {
+                logManager.createIfNotExists(streamName, settings.getPartitions(streamName));
+            } else {
                 int size = logManager.size(streamName);
                 if (settings.getPartitions(streamName) != size) {
                     log.debug(String.format(
                             "Update settings for stream: %s defined with %d partitions but exists with %d partitions",
                             streamName, settings.getPartitions(streamName), size));
                     settings.setPartitions(streamName, size);
                 }
-            } else {
-                logManager.createIfNotExists(streamName, settings.getPartitions(streamName));
             }
-            streams.add(streamName);
```

Comment:
Adding the stream isn't needed anymore?

Reply:
The streams list was not used so it is removed yes.